File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/monitoring/config/HttpServerMetricsFilterConfiguration.java
package com.study.webflux.rag.infrastructure.monitoring.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import io.micrometer.core.instrument.config.MeterFilter;

/**
 * 운영 트래픽 집계에서 actuator 요청 메트릭을 제외합니다.
 */
@Configuration
public class HttpServerMetricsFilterConfiguration {

	@Bean
	public MeterFilter actuatorRequestMetricsDenyFilter() {
		return MeterFilter.deny(id -> {
			if (!"http.server.requests".equals(id.getName())) {
				return false;
			}

			String uri = id.getTag("uri");
			return uri != null && uri.startsWith("/actuator");
		});
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/monitoring/config/HttpServerMetricsFilterConfiguration.kt
package com.miyou.app.infrastructure.monitoring.config

import io.micrometer.core.instrument.config.MeterFilter
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * actuator endpoint 접근을 제외한 actuator 요청 metrics 수집을 제외하기 위한 설정입니다.
 */
@Configuration
class HttpServerMetricsFilterConfiguration {
    @Bean
    fun actuatorRequestMetricsDenyFilter(): MeterFilter =
        MeterFilter.deny { id ->
            if (id.name != "http.server.requests") {
                false
            } else {
                id.getTag("uri")?.startsWith("/actuator") == true
            }
        }
}
